File: auth/authentication/core/src/main/java/io/ehdev/conrad/authentication/cookie/UserCookieManger.java
package io.ehdev.conrad.authentication.cookie;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

public interface UserCookieManger {

    void addCookie(String contents, HttpServletResponse response);

    void removeCookie(HttpServletResponse response);

    String readCookieValue(HttpServletRequest request);
}

File: internal/authentication/src/main/kotlin/tech/crom/security/authentication/cookie/UserCookieManger.kt
package tech.crom.security.authentication.cookie

import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

interface UserCookieManger {

    fun addCookie(contents: String, response: HttpServletResponse)

    fun removeCookie(response: HttpServletResponse)

    fun readCookieValue(request: HttpServletRequest): String?
}
